swo_mpt_api/billing/journal_client.py

```diff
@@ -94,10 +94,39 @@ def update(self, journal_id, journal: Journal) -> Journal:
         response.raise_for_status()
         return response.json()
 
-    def upload(self, journal_id, journals_file: IO):
-        response = self._client.post(
-            f"/billing/journals/{journal_id}/upload", files={"file": journals_file}
-        )
+    def upload(
+        self, journal_id, file: IO, filename: str | None = None, file_type: str | None = None
+    ) -> Journal:
+        """
+        Uploads a file associated with a specific journal.
+
+        This method is used to upload a file to a specified journal using the given journal
+        ID. The file is sent to the appropriate endpoint, which processes it as part of the
+        specified journal. You can provide an optional filename and file type. If not
+        provided, defaults for these values will be used.
+
+        Parameters:
+            journal_id: The unique identifier of the journal to which the file will be
+                uploaded.
+            file: A file-like object (supporting read operations) to be uploaded.
+            filename: An optional string specifying the name of the file. Defaults to
+                the name attribute of the file object if not provided.
+            file_type: An optional string representing the MIME type of the file.
+                Defaults to "application/jsonl" if not provided.
+
+        Returns:
+            Dict: A dictionary containing the response data from the upload operation.
```

Comment:
We can avoid specifying the type here and just write -> A dictionary....or maybe it's better to write -> A journal containing....